tests/conjectures_tests.py

```diff
@@ -0,0 +1,101 @@
+import unittest
+from ramanujan.LHSHashTable import LHSHashTable
+from ramanujan.enumerators.EfficientGCFEnumerator import EfficientGCFEnumerator
+from ramanujan.poly_domains.CartesianProductPolyDomain import CartesianProductPolyDomain
+from ramanujan.poly_domains.Zeta3Domain1 import Zeta3Domain1
+from ramanujan.constants import g_const_dict
+
+
+def get_testable_data(refined_res_list):
+    '''
+        On those test we have no use for the bloom key or lhs dict pointers
+        exctact the meaning full data for testing
+    '''
+    data = []
+    for i in refined_res_list:
+        data.append((
+            i.rhs_an_poly,
+            i.rhs_bn_poly,
+            i.c_top,
+            i.c_bot))
+    return data
+
+
+class APITests(unittest.TestCase):
+
+    def test_MITM_api1(self):
+        saved_hash = 'e_lhs_dept5_db'
+        lhs_search_limit = 5
+        lhs = LHSHashTable(saved_hash, lhs_search_limit, [g_const_dict['e']]) 
+
+        poly_search_domain = CartesianProductPolyDomain(
+            2, [-5, 5],
+            2, [-5, 5])
+
+        enumerator = EfficientGCFEnumerator(lhs, poly_search_domain, [g_const_dict['e']])
+
+        results = get_testable_data(enumerator.full_execution())
+
+        self.assertEqual(len(results), 42)
+        self.assertIn(
+            ((0, 4, 2), (0, 0, 1), (1, 1), (-1, 1)),
+            results)
+        self.assertIn(
+            ((0, 1, 1), (0, 1, 0), (1, 0), (-2, 1)),
+            results)
+
+    def test_MITM_api2(self):
+        saved_hash = 'zeta3.lhs.dept14.db'
+        lhs_search_limit = 14
+        lhs = LHSHashTable(saved_hash, lhs_search_limit, [g_const_dict['zeta'](3)]) 
+
+        poly_search_domain = Zeta3Domain1(
+            [(2, 2), (1, 1), (1, 17), (1, 5)], # an coefs
+            (-16, -1) # bn coef
+            ) 
+
+        enumerator = EfficientGCFEnumerator(
+            lhs,
+            poly_search_domain,
+            [g_const_dict['zeta'](3)]
+            )
+
+        results = get_testable_data(enumerator.full_execution())
+
+        self.assertEqual(len(results), 3)
+        self.assertIn(
+            ((2, 1, 3, 1), (-1,), (8, 0), (0, 7)), 
+            results)
+        self.assertIn(
+            ((2, 1, 5, 2), (-16,), (12, 0), (0, 7)), 
+            results)
+        self.assertIn(
+            ((2, 1, 17, 5), (-1,), (6, 0), (0, 1)),
+            results)
+
+    def test_MITM_api4(self):
```

Comment:
name jumps from api2 to api4.
you can leave it like this if you promise to add api3 (i think it was Catalan?)

Reply:
updated it to be in the correct order. On the next PR after I'll add catalan's domain ill add the missing test